File: samples/snippets/cpp/VS_Snippets_CLR/ObjDispEx/CPP/objdispexc.cpp

//<Snippet1>
using namespace System;
using namespace System::IO;
int main()
{
   MemoryStream^ ms = gcnew MemoryStream( 16 );
   ms->Close();
   try
   {
      ms->ReadByte();
   }
   catch ( ObjectDisposedException^ e ) 
   {
      Console::WriteLine( "Caught: {0}", e->Message );
   }

}

//</Snippet1>

File: samples/snippets/csharp/VS_Snippets_CLR/ObjDispEx/CS/objdispexc.cs
//<Snippet1>
using System;
using System.IO;

public class ObjectDisposedExceptionTest 
{
   public static void Main()
   {     
      MemoryStream ms = new MemoryStream(16);
      ms.Close();
      try 
      {
         ms.ReadByte();
      } 
      catch (ObjectDisposedException e) 
      {
         Console.WriteLine("Caught: {0}", e.Message);
      }
   }
}
//</Snippet1>